### PR TITLE
#410 Working on Access plugin for Organic Groups

### PR DESCRIPTION
**Pull request by cmcintosh**
Placeholder PR until I'm ready for a full PR back into the project.

### CODE REVIEW

**Comment by cmcintosh** [comment]
Currently, I have some issues I am working through on this. Specifically getting the entity_type for the GroupCheck access callback.  Atm views arguments are being passed as strings, in the GroupCheck access function, it expects it to be a group already.

File: src/Access/GroupCheck.php [hunk 1]
```diff
@@ -84,6 +84,10 @@ public function access(AccountInterface $user, Route $route, RouteMatchInterface
         return AccessResult::forbidden();
       }
 
+      if(is_string($group)) {
```
Comment (MPParsley) [inline src/Access/GroupCheck.php]:
```suggestion
      if (is_string($group)) {
```

Comment (MPParsley) [inline src/Access/GroupCheck.php:86]:
```suggestion
      if (is_string($group)) {
```

File: src/Plugin/views/access/GroupCheck.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+*
+* @ingroup views_access_plugins
+*
+* @ViewsAccess(
+*   id = "group_check",
+*   title = @Translation("Group Check"),
+*   help = @Translation("Access will be granted to users with memberships to the audience for this view. The audience will be determine by the route.")
+* )
+*/
+class GroupAudience extends AccessPluginBase {
```
Comment (MPParsley) [inline src/Plugin/views/access/GroupCheck.php]:
```suggestion
/**
 * Access plugin that provides group-audience based access control.
 *
 * @ingroup views_access_plugins
 *
 * @ViewsAccess(
 *   id = "group_check",
 *   title = @Translation("Group Check"),
 *   help = @Translation("Access will be granted to users with memberships to the audience for this view. The audience will be determine by the route.")
 * )
 */
class GroupAudience extends AccessPluginBase {
```

File: src/Access/GroupCheck.php [hunk 3]
```diff
@@ -83,6 +83,10 @@ public function access(AccountInterface $user, Route $route, RouteMatchInterface
         return AccessResult::forbidden();
       }
 
+      if (is_string($group)) {
+        $group = entity_load($parameter_name, $group);
```
Comment (MPParsley) [inline src/Access/GroupCheck.php:89]:
```
FILE: /home/travis/build/Gizra/og/src/Access/GroupCheck.php
--------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
--------------------------------------------------------------------------------
 87 | WARNING | entity_load() calls should be avoided in classes, use
    |         | dependency injection and the "entity_type.manager" service
    |         | instead
--------------------------------------------------------------------------------

```

File: src/Plugin/views/access/GroupCheck.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\og\Plugin\views\access;
```
Comment (MPParsley) [inline src/Plugin/views/access/GroupCheck.php:5]:
```suggestion
declare(strict_types = 1);

namespace Drupal\og\Plugin\views\access;
```

**Review by MPParsley — CHANGES_REQUESTED** [review]
```
FILE: /home/travis/build/Gizra/og/src/Plugin/views/access/GroupCheck.php
--------------------------------------------------------------------------------
FOUND 47 ERRORS AND 13 WARNINGS AFFECTING 53 LINES
--------------------------------------------------------------------------------
  20 | ERROR   | [ ] Class name doesn't match filename; expected "class
     |         |     GroupCheck"
  42 | ERROR   | [x] Expected 3 space(s) before asterisk; 2 found
  43 | ERROR   | [x] Expected 3 space(s) before asterisk; 2 found
  44 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  45 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  47 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  49 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  50 | ERROR   | [x] Expected 4 space(s) before asterisk; 3 found
  51 | ERROR   | [x] Expected 4 space(s) before asterisk; 3 found
  52 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  53 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  54 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  55 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  56 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  57 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  58 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  60 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  61 | ERROR   | [x] Expected 4 space(s) before asterisk; 3 found
  62 | ERROR   | [x] Expected 4 space(s) before asterisk; 3 found
  63 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
  64 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  65 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  67 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  69 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
  70 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
  74 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  75 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
  75 | WARNING | [x] A comma should follow the last multiline array item.
     |         |     Found: )
  77 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  78 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  78 | ERROR   | [x] Expected 1 space after FOREACH keyword; 0 found
  79 | ERROR   | [x] Line indented incorrectly; expected 6 spaces, found 7
  80 | ERROR   | [x] Line indented incorrectly; expected 8 spaces, found 9
  81 | ERROR   | [x] Line indented incorrectly; expected 10 spaces, found 13
  82 | ERROR   | [x] Line indented incorrectly; expected 8 spaces, found 9
  83 | ERROR   | [x] Line indented incorrectly; expected 6 spaces, found 7
  84 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  85 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  87 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
  88 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
  93 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  93 | ERROR   | [x] There should be no white space after an opening "["
  93 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
  94 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  94 | WARNING | [ ] Variable $groupEntitySelect is undefined.
  94 | WARNING | [ ] Variable $groupEntitySelect is undefined.
  95 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  96 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  97 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  98 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
  99 | ERROR   | [x] Line indented incorrectly; expected 6 spaces, found 7
 100 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
 102 | ERROR   | [x] Line indented incorrectly; expected 4 spaces, found 5
 104 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
 105 | WARNING | [ ] t() calls should be avoided in classes, use
     |         |     \Drupal\Core\StringTranslation\StringTranslationTrait and
     |         |     $this->t() instead
 109 | WARNING | [x] A comma should follow the last multiline array item.
     |         |     Found: 'edit-options-wrapper'
 110 | WARNING | [x] A comma should follow the last multiline array item.
     |         |     Found: ]
 112 | ERROR   | [x] Line indented incorrectly; expected 2 spaces, found 3
 112 | ERROR   | [x] Expected 1 blank line after function; 0 found
 113 | ERROR   | [x] The closing brace for the class must have an empty line
     |         |     before it
--------------------------------------------------------------------------------
PHPCBF CAN FIX THE 49 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------
```

**Comment by claudiu-cristea** [comment]
Since https://www.drupal.org/project/og/issues/3467694 was merged into the `8.x-1.x` branch, the development has been moved to https://www.drupal.org/project/og. If this change is still actual, please open an issue at https://www.drupal.org/project/issues/og and provide there a _merge request_ similar with this pull request.

Thank you for contributing to Organic groups module.

**Pull request closed without merging**